scipy/sparse/linalg/_expm_multiply.py

```diff
@@ -595,7 +598,8 @@ def _expm_multiply_interval_core_0(A, X, h, mu, m_star, s, q):
     A helper function, for the case q <= s.
     """
     for k in range(q):
-        X[k+1] = _expm_multiply_simple_core(A, X[k], h, mu, m_star, s)
+        s_step = -(-s//q)  # Ceiling division
```

Comment:
1) Is this a recipe in Higham's paper?
2) The comment here needs to explain the reasoning why this is correct.
3) The time step scales as `t/q` but doesn't the scaling scale with `1/2^s`, so is the linear scaling here ok? 
4) Should the per-step s be computed instead with fragment_3_1, before calling core_0 around line 578?